foundation/foundation-core-java/src/main/java/com/boozallen/aissemble/core/filestore/AbstractFileStore.java

```diff
@@ -45,10 +45,14 @@ public AbstractFileStore(final String name) {
      * @return a fully connected context
      */
     public BlobStoreContext openContext() {
-        return ContextBuilder.newBuilder(fileStoreConfig.getProvider())
-                .credentials(fileStoreConfig.getAccessKeyId(), fileStoreConfig.getSecretAccessKey())
-                .overrides(fileStoreConfig.getOverrides())
-                .buildView(BlobStoreContext.class);
+        ContextBuilder contextBuilder = ContextBuilder.newBuilder(fileStoreConfig.getProvider())
+                .credentials(fileStoreConfig.getAccessKeyId(), fileStoreConfig.getSecretAccessKey());
+
+        if (!fileStoreConfig.getOverrides().isEmpty()) {
```

Comment:
Q: Why is this change necessary? Does an empty overrides map cause issues? I'd think since the getter never returns `null` then it'd be safe to just pass it in.

Reply:
Yes, when the Overrides map is empty, we are getting a NPE and it can't invoke the map. To avoid the NPE from being thrown, the logic will now check to see if that map is null, if its not then it will chain it to the `ContextBuilder`.

Reply:
@habibimoiz I don't think we actually tested with the empty map, we only tested with a null map. It might be fine to always add the map now that we guarantee `.getOverrides()` will never return null.

Reply:
This actually ends up causing a new NPE if the contract for `getOverrides` changes and `null` is sometimes returned. FWIW, `ContextBuilder` would throw a NPE if `getOverrides` did return `null`. So we could null-guard this instead of calling `isEmpty`. Or we could just clarify in the JavaDocs of `getOverrides` that it returns an empty map if the value hasn't been provided.

Reply:
I am in favor of null-guarding it. 

Proposed change: 
```
    public BlobStoreContext openContext() {
        ContextBuilder contextBuilder = ContextBuilder.newBuilder(fileStoreConfig.getProvider())
                .credentials(fileStoreConfig.getAccessKeyId(), fileStoreConfig.getSecretAccessKey());

        if (fileStoreConfig.getOverrides() != null) {
            contextBuilder.overrides(fileStoreConfig.getOverrides());
        }
        return contextBuilder.buildView(BlobStoreContext.class);
    }
```

Reply:
Let's verify is still works with an empty overrides as well.

Reply:
Just tested out with both null and empty overrides and I am getting the expected outcome!